docs/src/main/asciidoc/ap4k.adoc

```diff
@@ -0,0 +1,155 @@
+= {project-name} - Deploying an Application to Kubernetes and OpenShift
+
+This guide covers:
+
+* The deployment of the application to Kubernetes
+* The deployment of the application to OpenShift
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* roughly 15 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+* access to Kubernetes or Openshift cluster (Minikube and Minishift are both viable options)
+* have the following binaries on your PATH: `docker` (for building and pushing images), `kubectl` (for deploying to Kubernetes) and `oc` (for deploying to Openshift)
+
+Remember, you need to configure Maven as indicated in the link:maven-config.html[Maven configuration page].
+
+== Creating the Maven project
+
+First, we need a new project that contains the ap4k extension. This can be done using the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:${quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=ap4k \
+    -DclassName="org.acme.rest.GreetingResource" \
+    -Dpath="/greeting" \
+    -Dextensions="ap4k"
+----
+
+With project setup, proceed to create a `src/main/java/org/acme/rest/MyApplication.java` file that contains the following source:
+
+[source,java]
+----
+import javax.ws.rs.core.Application;
+
+public class MyApplication extends Application {
```

Comment:
Apparently, you get a RESTEasy deprecation warning if you do so? (connecting dots from a discussion on Zulip)

Reply:
LOL :P